app/Metadata/Extractor.php

```diff
@@ -175,16 +174,52 @@ public function extract(string $filename, string $type): array
 		$metadata['livePhotoContentID'] = ($exif->getContentIdentifier() !== false) ? $exif->getContentIdentifier() : null;
 		$metadata['MicroVideoOffset'] = ($exif->getMicroVideoOffset() !== false) ? $exif->getMicroVideoOffset() : null;
 
-		// We need to make sure, takestamp is between '1970-01-01 00:00:01' UTC to '2038-01-19 03:14:07' UTC.
-		// We set value to null in case we're out of bounds
-		if ($metadata['takestamp'] !== null) {
+		$takestamp = $exif->getCreationDate();
+		if ($takestamp !== false) {
+			// Some videos use local time while others use UTC and it's
+			// often impossible to tell.  We rely here on a simple
+			// filetype-based heuristics and, if needed, convert a UTC
+			// timestamp to the local time of the Lychee server so that
+			// it's displayed to users as local time.
+			//
+			// Other possible approaches would include deriving local time from
+			// the file name or from other objects in the same album, as well
+			// as extracting the time zone from the location data if present.
+			if ($kind === 'video') {
+				$locals = strtolower(Configs::get_value('local_takestamp_video_formats', ''));
+				if (!in_array(strtolower($extension), explode('|', $locals), true)) {
+					// This is a video format where we expect the takestamp
+					// to be provided in UTC .
+					if ($takestamp->getTimezone()->getName() === date_default_timezone_get()) {
```

Comment:
Can you help me on this one? I was assuming that the time zone is set to 'Zulu' if it's not set in the exifdata.

Reply:
Without having tested it, I think it would screw up my videos shot with my iPhone in my time zone, wouldn't it?
The iPhone shoots mov files and adds the correct time zone (which is equal to my server time zone).

Reply:
This is indeed a tricky part of the code that involves certain subtleties that I didn't discuss in the comments.

Depending on the file type and the metadata extractor used, the timestamp they output either will have a `Z` at the end or not. There is no clear-cut rule; e.g., FFprobe adds a `Z` for mov and mp4 files but Exiftool does not, though both do it for webm files and neither does it for avi files.

So when the extracted timestamp string gets passed to `DateTime`, it will either get interpreted as UTC time (if it had a `Z` at the end) or as a local time (if it didn't). It's that second case that the code is dealing with in the line you questioned.

Now the second question you asked is what if the video file _had_ the time zone embedded, as mov files from iPhones do? If the time zone on the server is the same as in your phone, wouldn't that wrongly trigger the above condition?

I worried about that possibility as well but testing shows that it works fine. That's because `date_default_timezone_get` returns strings such as `UTC` or, in my case, `America/Chicago`, whereas `getTimezone()->getName()` for iPhone files returns `-05:00`. So even though they indicate the same time zone, the strings are different, the comparison fails, and we don't do anything.

Now, given the length of this explanation, I think you will understand why I didn't put this in the comments :smile:.

Reply:
omg. If that's the case, then I'm fine